pulsar-client/pom.xml

```diff
@@ -114,6 +114,11 @@
       <artifactId>gson</artifactId>
     </dependency>
 
+    <dependency>
+      <groupId>org.apache.httpcomponents</groupId>
+      <artifactId>httpclient</artifactId>
+    </dependency>
```

Comment:
If we add a new dependency, we'd need to also shade it. It would be better if we can use the Java default HTTP client

Reply:
I used the `AsyncHttpClient` instead of `org.apache.httpcomponents.httpclient`, avoiding introduce other dependency.